devicetypes/smartthings/orvibo-Moisture-Sensor.src/orvibo-Moisture-Sensor.groovy

```diff
@@ -0,0 +1,192 @@
+			state "both", label:'alarm!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+		}
+
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "water"
+		details(["water", "alarm", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	log.debug "description: $description"
+
+	def result = [:]
+	Map map = zigbee.getEvent(description)
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			ZoneStatus zs = zigbee.parseZoneStatus(description)
+			map = zs.isAlarm1Set() ? getMoistureResult('wet') : getMoistureResult('dry')
+			zs.isAlarm1Set()?sendEvent(name: "alarm", value: "siren"):sendEvent(name: "alarm", value: "off")
+			result = createEvent(map)
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+			if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+				def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+				map = zs.isAlarm1Set() ? getMoistureResult('wet') : getMoistureResult('dry')
+				result = createEvent(map)
+			} else if (descMap?.clusterInt == 0x0001 && descMap?.commandInt != 0x07 && descMap?.value) {
+				if(descMap?.attrInt == 0x0021){
+					map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+					result = createEvent(map)
+				}else if(descMap?.attrInt == 0x0020){
+					map = getBatteryResult(Integer.parseInt(descMap.value, 16))
+					result = createEvent(map)
+				}
+			}
+		}
+	}
+
+	log.debug "Parse returned $result"
+
+	result
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	//the device support two way to get the battery value
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021)
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020)
+	refreshCmds += zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+		zigbee.enrollResponse()
+
+	refreshCmds
+}
+
+def configure() {
+	sendEvent(name: "checkInterval", value: 20 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "Configuring Reporting"
+	def configCmds = []
+	configCmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021, DataType.UINT8, 30, 21600, 0x10)
+	//	configCmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020, DataType.UINT8, 30, 21600, 0x10)
```

Comment:
if we don't configure the reporting of this cluster, will we ever be receiving unsolicited reports using it?

Reply:
There is no min & max operating voltages for this device.So just the 0x0021 attribute can be used for getting battery value, I will delete it .